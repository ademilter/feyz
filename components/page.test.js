import { render, screen } from '@testing-library/react'
import Page from './page'
import getData from '../utils/get-data'

describe('Page', () => {
  test('should render correctly.', async () => {
    const { data, totalData } = await getData()

    const { asFragment } = render(
      <Page pathname={'/'} data={data} totalData={totalData} />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
