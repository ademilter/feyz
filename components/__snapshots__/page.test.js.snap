// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should have render. 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <h1
      class=""
    >
      <a
        href="/"
      >
        Feyz
      </a>
    </h1>
    <div
      class=""
      data-testid="nav"
    >
      <a
        href="/tag/tasarim"
      >
        Tasarım
      </a>
      <a
        href="/tag/frontend"
      >
        Frontend
      </a>
      <a
        href="/tag/yazilim"
      >
        Yazılım
      </a>
      <a
        href="/tag/turkce-kaynak"
      >
        Türkçe Kaynak
      </a>
      <a
        href="/tag/video"
      >
        Video
      </a>
      <a
        href="/tag/podcast"
      >
        Podcast
      </a>
      <a
        href="/tag/yazi"
      >
        Blog Yazısı
      </a>
      <a
        href="/tag/arac"
      >
        Araç
      </a>
      <a
        href="/tag/etkinlik"
      >
        Etkinlik
      </a>
      <a
        href="/tag/alinti"
      >
        Alıntı
      </a>
      <a
        class=""
        href="https://www.patreon.com/ademilter"
        rel="noopener noreferrer"
        target="_blank"
      >
        Destek Ol
      </a>
    </div>
    <div
      class=""
      data-testid="news"
    >
      <article
        class=""
      >
        <figure>
          <a
            aria-hidden="true"
            class=""
            href="https://chadd.net"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            <picture>
              <source
                srcset="http://lorempixel.com/913/512"
              />
              <img
                alt="Doloremque animi officia."
                loading="lazy"
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVQYV2P4DwABAQEAWk1v8QAAAABJRU5ErkJggg=="
              />
            </picture>
          </a>
        </figure>
        <div>
          <header>
            <h2>
              <a
                class=""
                href="https://chadd.net"
                rel="noopener noreferrer"
                target="_blank"
              >
                Doloremque animi officia.
              </a>
            </h2>
            <p>
              Eum facere ut expedita repudiandae iure labore laborum rerum. Dolorum autem ipsum similique iure error doloremque est. Hic voluptates assumenda nisi fugit dicta in.
            </p>
          </header>
          <footer>
            <a
              aria-hidden="true"
              class=""
              href="https://chadd.net"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              3 months ago
            </a>
            , 
            <a
              class=""
              href="https://twitter.com/Elliott9"
              rel="noopener noreferrer"
              target="_blank"
            >
              Elliott9
            </a>
            , 
            <a
              href="/tag/frontend"
            >
              Frontend
            </a>
            , 
            <a
              href="/tag/turkce-kaynak"
            >
              Türkçe Kaynak
            </a>
            , 
            <a
              href="/tag/podcast"
            >
              Podcast
            </a>
            , 
            <a
              href="/tag/etkinlik"
            >
              Etkinlik
            </a>
          </footer>
        </div>
      </article>
      <article
        class=""
      >
        <figure>
          <a
            aria-hidden="true"
            class=""
            href="https://mossie.com"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            <picture>
              <source
                srcset="http://lorempixel.com/913/512"
              />
              <img
                alt="Id facilis sed alias nulla omnis sed est."
                loading="lazy"
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVQYV2P4DwABAQEAWk1v8QAAAABJRU5ErkJggg=="
              />
            </picture>
          </a>
        </figure>
        <div>
          <header>
            <h2>
              <a
                class=""
                href="https://mossie.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Id facilis sed alias nulla omnis sed est.
              </a>
            </h2>
            <p>
              Non dicta delectus ipsam aliquam ipsam. Consequatur tenetur saepe magnam reprehenderit quis earum est. Et vel et ullam dolore eius suscipit in.
            </p>
          </header>
          <footer>
            <a
              aria-hidden="true"
              class=""
              href="https://mossie.com"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              4 months ago
            </a>
            , 
            <a
              class=""
              href="https://twitter.com/Newton21"
              rel="noopener noreferrer"
              target="_blank"
            >
              Newton21
            </a>
            , 
            <a
              href="/tag/frontend"
            >
              Frontend
            </a>
            , 
            <a
              href="/tag/yazi"
            >
              Blog Yazısı
            </a>
            , 
            <a
              href="/tag/yazilim"
            >
              Yazılım
            </a>
            , 
            <a
              href="/tag/etkinlik"
            >
              Etkinlik
            </a>
            , 
            <a
              href="/tag/arac"
            >
              Araç
            </a>
          </footer>
        </div>
      </article>
      <article
        class=""
      >
        <h2>
          <a
            class=""
            href="http://alysha.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Unde sed at. Sed et velit consequatur sed.
          </a>
        </h2>
        <footer>
          <a
            aria-hidden="true"
            class=""
            href="http://alysha.com"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            11 months ago
          </a>
          , 
          <a
            class=""
            href="https://twitter.com/Blaise_Streich"
            rel="noopener noreferrer"
            target="_blank"
          >
            Blaise_Streich
          </a>
          , 
          <a
            href="/tag/yazilim"
          >
            Yazılım
          </a>
          , 
          <a
            href="/tag/turkce-kaynak"
          >
            Türkçe Kaynak
          </a>
          , 
          <a
            href="/tag/alinti"
          >
            Alıntı
          </a>
          , 
          <a
            href="/tag/yazi"
          >
            Blog Yazısı
          </a>
          , 
          <a
            href="/tag/podcast"
          >
            Podcast
          </a>
        </footer>
      </article>
      <article
        class=""
      >
        <h2>
          <a
            class=""
            href="http://malachi.net"
            rel="noopener noreferrer"
            target="_blank"
          >
            Nesciunt asperiores animi itaque nesciunt ut optio aliquam mollitia aut. Dolores repellat nostrum eveniet. Molestiae perferendis sed cupiditate.
          </a>
        </h2>
        <footer>
          <a
            aria-hidden="true"
            class=""
            href="http://malachi.net"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            5 months ago
          </a>
          , 
          <a
            class=""
            href="https://twitter.com/Devon_Kohler"
            rel="noopener noreferrer"
            target="_blank"
          >
            Devon_Kohler
          </a>
          , 
          <a
            href="/tag/yazilim"
          >
            Yazılım
          </a>
          , 
          <a
            href="/tag/tasarim"
          >
            Tasarım
          </a>
          , 
          <a
            href="/tag/alinti"
          >
            Alıntı
          </a>
          , 
          <a
            href="/tag/arac"
          >
            Araç
          </a>
          , 
          <a
            href="/tag/yazi"
          >
            Blog Yazısı
          </a>
        </footer>
      </article>
      <article
        class=""
      >
        <h2>
          <a
            class=""
            href="http://alexandrea.name"
            rel="noopener noreferrer"
            target="_blank"
          >
            Quam voluptatem beatae eum velit corporis. Unde at quasi enim sapiente distinctio quas aut et labore.
          </a>
        </h2>
        <footer>
          <a
            aria-hidden="true"
            class=""
            href="http://alexandrea.name"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            8 months ago
          </a>
          , 
          <a
            class=""
            href="https://twitter.com/Vivian_Beier"
            rel="noopener noreferrer"
            target="_blank"
          >
            Vivian_Beier
          </a>
          , 
          <a
            href="/tag/arac"
          >
            Araç
          </a>
          , 
          <a
            href="/tag/alinti"
          >
            Alıntı
          </a>
          , 
          <a
            href="/tag/frontend"
          >
            Frontend
          </a>
          , 
          <a
            href="/tag/podcast"
          >
            Podcast
          </a>
          , 
          <a
            href="/tag/yazilim"
          >
            Yazılım
          </a>
        </footer>
      </article>
      <article
        class=""
      >
        <figure>
          <a
            aria-hidden="true"
            class=""
            href="https://demetrius.name"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            <picture>
              <source
                srcset="http://lorempixel.com/913/512"
              />
              <img
                alt="Vel magni rerum ipsam facere velit voluptas doloribus doloremque."
                loading="lazy"
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVQYV2P4DwABAQEAWk1v8QAAAABJRU5ErkJggg=="
              />
            </picture>
          </a>
        </figure>
        <div>
          <header>
            <h2>
              <a
                class=""
                href="https://demetrius.name"
                rel="noopener noreferrer"
                target="_blank"
              >
                Vel magni rerum ipsam facere velit voluptas doloribus doloremque.
              </a>
            </h2>
            <p>
              Fuga perferendis autem odit. Et qui possimus aut excepturi ut porro ipsum. Illo esse repellendus voluptate et doloribus ad vel.
            </p>
          </header>
          <footer>
            <a
              aria-hidden="true"
              class=""
              href="https://demetrius.name"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              2 months ago
            </a>
            , 
            <a
              class=""
              href="https://twitter.com/Dayana_Feest40"
              rel="noopener noreferrer"
              target="_blank"
            >
              Dayana_Feest40
            </a>
            , 
            <a
              href="/tag/video"
            >
              Video
            </a>
            , 
            <a
              href="/tag/turkce-kaynak"
            >
              Türkçe Kaynak
            </a>
          </footer>
        </div>
      </article>
      <article
        class=""
      >
        <h2>
          <a
            class=""
            href="http://mylene.name"
            rel="noopener noreferrer"
            target="_blank"
          >
            Velit eum magni voluptates eaque soluta aliquam eos porro. Et praesentium accusamus dolor dolorem ea voluptas a repellat et. Sapiente cupiditate magnam animi pariatur voluptatem quas sit. Accusantium et et officiis rerum et corporis. Id autem nam veniam voluptatem.
          </a>
        </h2>
        <footer>
          <a
            aria-hidden="true"
            class=""
            href="http://mylene.name"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            12 days ago
          </a>
          , 
          <a
            class=""
            href="https://twitter.com/Cristina.Harvey"
            rel="noopener noreferrer"
            target="_blank"
          >
            Cristina.Harvey
          </a>
          , 
          <a
            href="/tag/alinti"
          >
            Alıntı
          </a>
          , 
          <a
            href="/tag/etkinlik"
          >
            Etkinlik
          </a>
          , 
          <a
            href="/tag/podcast"
          >
            Podcast
          </a>
          , 
          <a
            href="/tag/tasarim"
          >
            Tasarım
          </a>
        </footer>
      </article>
      <article
        class=""
      >
        <figure>
          <a
            aria-hidden="true"
            class=""
            href="http://thalia.net"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            <picture>
              <source
                srcset="http://lorempixel.com/913/512"
              />
              <img
                alt="Dicta sunt sit rerum."
                loading="lazy"
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVQYV2P4DwABAQEAWk1v8QAAAABJRU5ErkJggg=="
              />
            </picture>
          </a>
        </figure>
        <div>
          <header>
            <h2>
              <a
                class=""
                href="http://thalia.net"
                rel="noopener noreferrer"
                target="_blank"
              >
                Dicta sunt sit rerum.
              </a>
            </h2>
            <p>
              Et aut vel provident voluptas itaque ad possimus eos quam. Nobis quas reiciendis consectetur suscipit quis quasi deleniti esse eos. Sunt nulla adipisci culpa tempore. Est et consequatur et magnam. Beatae atque et modi voluptatem. Omnis quia rerum nobis qui officia.
            </p>
          </header>
          <footer>
            <a
              aria-hidden="true"
              class=""
              href="http://thalia.net"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              4 months ago
            </a>
            , 
            <a
              class=""
              href="https://twitter.com/Gunner94"
              rel="noopener noreferrer"
              target="_blank"
            >
              Gunner94
            </a>
            , 
            <a
              href="/tag/arac"
            >
              Araç
            </a>
            , 
            <a
              href="/tag/yazilim"
            >
              Yazılım
            </a>
          </footer>
        </div>
      </article>
      <article
        class=""
      >
        <figure>
          <a
            aria-hidden="true"
            class=""
            href="http://taryn.com"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            <picture>
              <source
                srcset="http://lorempixel.com/913/512"
              />
              <img
                alt="Et vel et repellendus voluptatem molestiae explicabo quos minus."
                loading="lazy"
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVQYV2P4DwABAQEAWk1v8QAAAABJRU5ErkJggg=="
              />
            </picture>
          </a>
        </figure>
        <div>
          <header>
            <h2>
              <a
                class=""
                href="http://taryn.com"
                rel="noopener noreferrer"
                target="_blank"
              >
                Et vel et repellendus voluptatem molestiae explicabo quos minus.
              </a>
            </h2>
            <p>
              Quos eveniet quibusdam amet velit reprehenderit vel repellendus. Velit molestiae blanditiis quasi. Autem dolorum aspernatur qui non qui placeat maxime.
            </p>
          </header>
          <footer>
            <a
              aria-hidden="true"
              class=""
              href="http://taryn.com"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              10 months ago
            </a>
            , 
            <a
              class=""
              href="https://twitter.com/Zita.Lehner"
              rel="noopener noreferrer"
              target="_blank"
            >
              Zita.Lehner
            </a>
            , 
            <a
              href="/tag/video"
            >
              Video
            </a>
            , 
            <a
              href="/tag/tasarim"
            >
              Tasarım
            </a>
            , 
            <a
              href="/tag/turkce-kaynak"
            >
              Türkçe Kaynak
            </a>
            , 
            <a
              href="/tag/yazilim"
            >
              Yazılım
            </a>
            , 
            <a
              href="/tag/arac"
            >
              Araç
            </a>
          </footer>
        </div>
      </article>
      <article
        class=""
      >
        <figure>
          <a
            aria-hidden="true"
            class=""
            href="https://sophie.biz"
            rel="noopener noreferrer"
            tabindex="-1"
            target="_blank"
          >
            <picture>
              <source
                srcset="http://lorempixel.com/913/512"
              />
              <img
                alt="Molestiae et rerum laudantium aspernatur voluptatibus quas deleniti."
                loading="lazy"
                src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVQYV2P4DwABAQEAWk1v8QAAAABJRU5ErkJggg=="
              />
            </picture>
          </a>
        </figure>
        <div>
          <header>
            <h2>
              <a
                class=""
                href="https://sophie.biz"
                rel="noopener noreferrer"
                target="_blank"
              >
                Molestiae et rerum laudantium aspernatur voluptatibus quas deleniti.
              </a>
            </h2>
            <p>
              Ab et minus magni dolor suscipit. Molestiae qui nemo non ad. Est ex commodi.
            </p>
          </header>
          <footer>
            <a
              aria-hidden="true"
              class=""
              href="https://sophie.biz"
              rel="noopener noreferrer"
              tabindex="-1"
              target="_blank"
            >
              2 months ago
            </a>
            , 
            <a
              class=""
              href="https://twitter.com/Jo.Gislason47"
              rel="noopener noreferrer"
              target="_blank"
            >
              Jo.Gislason47
            </a>
            , 
            <a
              href="/tag/arac"
            >
              Araç
            </a>
            , 
            <a
              href="/tag/video"
            >
              Video
            </a>
            , 
            <a
              href="/tag/yazilim"
            >
              Yazılım
            </a>
          </footer>
        </div>
      </article>
    </div>
    <div
      class=""
      data-testid="pagination"
    >
      <a
        href="/"
      >
        1
      </a>
      <a
        href="/2"
      >
        2
      </a>
      <a
        href="/3"
      >
        3
      </a>
      <a
        href="/4"
      >
        4
      </a>
      <a
        href="/5"
      >
        5
      </a>
    </div>
  </div>
</DocumentFragment>
`;
